src/hotspot/share/oops/cpCache.cpp

```diff
@@ -672,18 +642,29 @@ void ConstantPoolCacheEntry::verify(outputStream* st) const {
 
 ConstantPoolCache* ConstantPoolCache::allocate(ClassLoaderData* loader_data,
                                      const intStack& index_map,
-                                     const intStack& invokedynamic_index_map,
-                                     const intStack& invokedynamic_map, TRAPS) {
+                                     const intStack& invokedynamic_map,
+                                     const GrowableArray<ResolvedIndyEntry> indy_entries,
+                                     TRAPS) {
 
-  const int length = index_map.length() + invokedynamic_index_map.length();
+  const int length = index_map.length();
   int size = ConstantPoolCache::size(length);
 
+  // Initialize resolvedinvokedynamicinfo array with available data
+  Array<ResolvedIndyEntry>* array;
+  if (indy_entries.length()) {
+    array = MetadataFactory::new_array<ResolvedIndyEntry>(loader_data, indy_entries.length(), CHECK_NULL);
+    for (int i = 0; i < indy_entries.length(); i++) {
+      array->at_put(i, indy_entries.at(i));
+    }
+  } else {
+    array = nullptr;
+  }
+
   return new (loader_data, size, MetaspaceObj::ConstantPoolCacheType, THREAD)
-    ConstantPoolCache(length, index_map, invokedynamic_index_map, invokedynamic_map);
+    ConstantPoolCache(length, index_map, invokedynamic_map, array);
```

Comment:
I think it reads better if this line is indented to right after the open parenthesis.